app/views/layouts/_html_head.html.erb

```diff
@@ -1,16 +1,17 @@
 <head>
-  <meta charset="utf-8">
-  <meta http-equiv="X-UA-Compatible" content="IE=edge">
-  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+  <meta charset="utf-8" />
+  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
   <title>Sara Alert</title>
   <%= csrf_meta_tags %>
   <%= csp_meta_tag %>
   <script>
-    window.BASE_PATH = "<%= ADMIN_OPTIONS['base_path'] %>";
-    window.SENTRY_URL = "<%= ADMIN_OPTIONS['sentry_url'] %>";
-    window.SARA_VERSION = "<%= ADMIN_OPTIONS['version'] %>";
-    window.USER_EMAIL = "<%= current_user&.email || 'NONE' %>";
-    window.USER_ID = "<%= current_user&.id || 'NONE' %>";
+    // Because the ERB should use to_json in all <script> tags, some values must be HTML escaped.
+    window.BASE_PATH = "<%= ADMIN_OPTIONS['base_path'].to_json %>".replace(/&quot;/g, '');
```

Comment:
I saw you mention this explicitly in the PR description, but can you explain why the to_json was necessary in the first place?

Reply:
To protect against XSS. For example, user controls the email field and could create XSS vulnerability by changing their email to, say `alert(1)`, and later on in the JS we evaluate it.

Just one potential example.